pages/elementFiles/elementFiles.js

```diff
@@ -26,12 +29,16 @@ router.get('/*', function(req, res, next) {
     let elementFilesDir;
     if (res.locals.course) {
         // Files should be served from the course directory
-        elementFilesDir = path.join(res.locals.course.path, 'elements');
+        const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
+        chunks.ensureChunksForCourse(res.locals.course.id, {'type': 'elements'}, (err) => {
```

Comment:
Newer pages are using https://www.npmjs.com/package/express-async-handler to do this. For example: https://github.com/PrairieLearn/PrairieLearn/blob/57a2255d372f457bc6655b37b60e40bf1458c240/pages/workspace/workspace.js#L30

I like your `safeAsync` function for being easy to drop in for a short segment of code, rather than writing the whole route handler as async/await with `express-async-handler`.

Reply:
Filed an issue for this.

Reply:
Ah! TIL about `express-async-handler`! The implementation seems to be practically identical to my own, except it correctly handles the 4-argument form of an express handler. I think we should prefer that over my suggestion.